### PR TITLE
keepassxc: 2.2.4 -> 2.3.0

### PR DESCRIPTION
**Pull request by jonafato**
###### Motivation for this change

See the [changelog](https://github.com/keepassxreboot/keepassxc/blob/a06e85f6ceac39851ca694464f74c927dd953f1e/CHANGELOG#L1-L51).

Changes to options:

- `withKeePassBrowser`: New in version 2.3.0 and replaces the functionality provided by the KeePassHTTP plugin. Users should install the [KeePassXC-Browser extension](https://github.com/keepassxreboot/keepassxc-browser). Defaults to `true` to remain consistent with the old default for `withKeePassHTTP`.

- `withKeePassHTTP`: This feature is deprecated and disabled by default in the source repository. Default changed from `true` to `false`.

- `withKeePassNetworking`: Controls all network access code within KeePassXC. Defaults to `false` to match the source repository.

- `withKeePassSSHAgent`: Enables the SSH Agent plugin for use with [KeeAgent](https://lechnology.com/software

- Add `darwin` to `libargon2` platforms for MacOS support.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/keepassx/community.nix [hunk 1]
```diff
@@ -1,28 +1,51 @@
...
+, glibcLocales
+, libyubikey
+, yubikey-personalization
+, libXi
+, qtx11extras
+
+, withKeePassBrowser ? true
+, withKeePassHTTP ? false
+, withKeePassNetworking ? false
+, withKeePassSSHAgent ? false
```
Comment (Mic92) [inline pkgs/applications/misc/keepassx/community.nix]: You would not enable this?

Comment (jonafato) [reply pkgs/applications/misc/keepassx/community.nix]: I don't use the feature, so I don't have much of an opinion on this. I went with the [default](https://github.com/keepassxreboot/keepassxc/blob/a06e85f6ceac39851ca694464f74c927dd953f1e/CMakeLists.txt#L50), but I'm happy to change it if it's generally useful to others.

Comment (Mic92) [reply pkgs/applications/misc/keepassx/community.nix]: The default is not every useful. It also disable `WITH_XC_BROWSER`.

**Comment by drdaeman** [comment]
Thanks for this PR!

JFYI: I've had to manually specify keepassxc-proxy location in the settings to get KeePassXC-Browser detect the app. I'm really not sure, but it could be that something out there needs some manual tweaking to Nix/NixOS filesystem layout.

**Comment by Mic92** [comment]
Let's address the browser location in a different pull request.

**Comment by Mic92** [comment]
I you don't specify a path it defaults to `keepassxc-proxy`:

```console
$ cat /home/joerg/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json
{
    "allowed_extensions": [
        "keepassxc-browser@keepassxc.org"
    ],
    "description": "KeePassXC integration with native messaging support",
    "name": "org.keepassxc.keepassxc_browser",
    "path": "/nix/store/vrjh4h32mw8ffnjx5kgy6w3k08gwg1gr-keepassxc-2.3.0/bin/keepassxc-proxy",
    "type": "stdio"
}
```

However this path might be garbage collected eventually.
Therefore a patch to replace it by the out put of  `which keepassxc-proxy` might be sensible.

**Comment by Mic92** [comment]
The string that needs to replaced here:
https://github.com/keepassxreboot/keepassxc/blob/develop/src/browser/HostInstaller.cpp#L178

**Comment by drdaeman** [comment]
Never mind my comment above - I was wrong, it also works without specifying the path. Not sure why KeePassXC sometimes does not get detected unless I open the settings dialog, but don't think it's a packaging issue.

BTW, that C++ code looks correct to me - it takes `keepassxc`'s binary path (`QFileInfo(QCoreApplication::applicationFilePath()).absolutePath()`) and deduces `keepass-proxy` path from it assuming both are in the same directory. If "update manifest files" is enabled, package upgrades should work without issues.

**Comment by Mic92** [comment]
I wonder why updating the manifest is not set by default.

**Pull request merged**